File: src/main/java/biblioteca/Revista.java
package biblioteca;

public class Revista extends Material {
    private int numeroEdicion;
    private String periodicidad;

    public Revista(int numeroEdicion, String periodicidad) {
    this.numeroEdicion = numeroEdicion;
    this.periodicidad = periodicidad;
}


    @Override
    public String prestar(String dniLector, int dias) {
        // Las revistas solo se prestan por 7 días máximo
        if (estado != EstadoMaterial.DISPONIBLE)
            throw new IllegalStateException();
        if (dias < 1 || dias > 7)   // PRECONDICIÓN MÁS RESTRICTIVA
            throw new IllegalArgumentException("Revistas: máx 7 días");
        estado = EstadoMaterial.PRESTADO;
        return "Revista prestada por" + dias + " días";
    }


    @Override
    public void devolver() {
        if (estado != EstadoMaterial.PRESTADO)
            throw new IllegalStateException("No está prestada");
        estado = EstadoMaterial.DISPONIBLE;
    }
}

File: biblioteca/revista.py
# biblioteca/revista.py

from biblioteca.material import Material
from biblioteca.libro import EstadoMaterial


class Revista(Material):
    def __init__(self, numero_edicion: int, periodicidad: str,
                 titulo: str, autor: str, anio: int):
        super().__init__(titulo, autor, anio)  # ← valida el año
        self._numero_edicion = numero_edicion
        self._periodicidad = periodicidad

    def prestar(self, dni_lector: str, dias: int) -> str:
        if self._estado != EstadoMaterial.DISPONIBLE:
            raise IllegalStateException("No disponible")
        if dias < 1 or dias > 7:
            raise ValueError("Revistas: máx 7 días")
        self._estado = EstadoMaterial.PRESTADO
        return f'Revista prestada por {dias} días'

    def devolver(self) -> None:
        if self._estado != EstadoMaterial.PRESTADO:
            raise PermissionError("No está prestada")
        self._estado = EstadoMaterial.DISPONIBLE